streams/src/main/java/org/apache/kafka/streams/kstream/internals/CogroupedStreamAggregateBuilder.java

```diff
@@ -61,6 +65,24 @@
         groupedStreams.remove(kGrouped);
         kGrouped.ensureCopartitionWith(groupedStreams);
 
+        final Map<KGroupedStreamImpl<K, ?>, StreamsGraphNode> parentNodes = new LinkedHashMap<>();
+
+        for (final KGroupedStreamImpl<K, ?> repartitionReqs : groupPatterns.keySet()) {
+            parentNodes.put(repartitionReqs, repartitionReqs.streamsGraphNode);
+            if (repartitionReqs.repartitionRequired) {
+
+                final OptimizableRepartitionNodeBuilder<K, ?> repartitionNodeBuilder = optimizableRepartitionNodeBuilder();
+
+                createRepartitionSource(repartitionReqs.name, repartitionNodeBuilder, repartitionReqs.keySerde, repartitionReqs.valSerde);
```

Comment:
>Why do we pass in repartitionReqs.name as name-prefix unconditionally -- a user should have the option to specify a custom name similar to a "regular" grouping+aggregation.

Yes, I agree here as well.  We're passing the name of the node, which could be a user-provided name via a `Named` parameter, but we explicitly use `Grouped` for naming repartition topics.

Reply:
>Should we make GroupedStreamAggregateBuilder and CogroupedStreamAggregateBuilder aware of each other to reuse created repartitions nodes? It seems that GroupedStreamAggregateBuilder tries to reuse an existing repartitioning node in build() 

I _**think so**_.  We attempt to re-use the repartition topic node due to an edge condition (see my immediate comment below). We'd have to follow the same rules and test with some different topologies to see how it works.

>(frankly, not sure why, because the optimizer should merge multiple together anyway?):

As for why we reuse the repartition node, there's an edge condition that occurs _**if**_ the user has named the repartition topic and attempts to use the `GroupedStream` in multiple operations with optimizations turned off - c.f the description in https://issues.apache.org/jira/browse/KAFKA-7758

Reply:
Thanks for pointing to the ticket -- I remember this now -- something this should consider, too.